4 - AnimalShelter/Code/Result.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace AnimalShelter.Code
+{
+    public class Result   
+    {
+        public bool WasAdded { get; set; }
+        public Animal Animal { get; set; }
+        public string ErrorMessage { get; set; }
+
+        private Result(bool wasAdded, Animal animal, string errorMessage)
+        {
+            WasAdded = wasAdded;
+            Animal = animal;
+            ErrorMessage = errorMessage;
+        }
+
+        public Result(Animal animal) : this(true, animal, null)
+        {
+
+        }
+
+        public Result(string errorMessage) : this(false, null, errorMessage)
+        {
+
+        }
+
+    }
```

Comment:
Great use of the result pattern. Some things to note, the "state" of this result should NOT be able to be set outside of this class. Your setters should be private. A result is a snap shot of a operation, not something that should change over time.